src/inventory_ui.cpp

```diff
@@ -2190,12 +2191,34 @@ item_location inventory_pick_selector::execute()
             }
         } else if( input.action == "INVENTORY_FILTER" ) {
             set_filter();
+        } else if( input.action == "EXAMINE" ) {
+            const inventory_entry &selected = get_active_column().get_selected();
+            if( selected ) {
+                const item *sitem = selected.any_item().get_item();
+                action_examine( sitem );
+            }
         } else {
             on_input( input );
         }
     }
 }
 
+void inventory_selector::action_examine( const item *sitem )
+{
+    // Code below pulled from the action_examine function in advanced_inv.cpp
+    std::vector<iteminfo> vThisItem;
+    std::vector<iteminfo> vDummy;
+    sitem->info( true, vThisItem );
+    item_info_data data( sitem->tname(), sitem->type_name(), vThisItem, vDummy );
+    data.handle_scrolling = true;
+
+    int maxwidth = std::max( FULL_SCREEN_WIDTH, TERMX );
+    int width = std::min( 80, maxwidth );
+
+    draw_item_info( [&]() -> catacurses::window {
+        return catacurses::newwin( 0, width, point( width, 0 ) ); }, data ).get_first_input();
```

Comment:
This callback may be called again on window resize, so `maxwidth` and `width` should be calculated inside the callback.

Reply:
Thanks, will do! Tested if window resizing works; had to update the point() coordinates so the window actually gets placed in the middle upon resize as well. Should be working now.